src/bin/cli/mod.rs

```diff
@@ -515,20 +515,23 @@ impl PackageTrait for DocPackage {
 pub fn imports_arg<T: PackageTrait>(package: &T) -> FileResolver {
     let mut resolver = FileResolver::default();
 
-    for filename in package.get_input() {
-        if let Ok(path) = PathBuf::from(filename).canonicalize() {
-            let _ = resolver.add_import_path(path.parent().unwrap());
-        }
-    }
-
+    let mut added_path = false;
```

Comment:
This is kludgy, and we can improve the implementation. The problem is, there isn't an easy way to tell if import paths are empty (querying `file_resolver.get_import_path(0)` might work, but it's not very clear, I prefer this stylistically).

Happy to fix this if there is an API change

Reply:
I don't understand what an empty `""` path means. What does an import path of nothing do? I suspect it is the same as `"."`. We should be permitting an empty path, let alone adding one.

solc does not allow it either.
```bash
$ solc --include-path "" 
--include-path option requires a non-empty base path.
```

Reply:
We had discussed this before. an empty base-path (which is different than an empty include path, which can't exist), is the same as `"."` except for when you import an absolute path `import "/usr/node_modules/Contract.sol";`. If base-path is `"."` then this import will be expanded to `"./usr/node_modules/Contract.sol"`. It is horrible and cursed, but it's the solidity semantics.

As for solc, your example has two errors:
1. you can never specify an include path when you have an empty base path (which you have by default).
2. you can never specify an empty include path:

```
$ solc --base-path ""
No input files given. If you wish to use the standard input please specify "-" explicitly.
$ solc --base-path . --include-path ""
Empty values are not allowed in --include-path.
```

[Solidity docs](https://docs.soliditylang.org/en/v0.8.17/path-resolution.html#base-path-and-include-paths) are explicit:
> _"By default the base path is empty, which leaves the source unit name unchanged. When the source unit name is a relative path, this results in the file being looked up in the directory the compiler has been invoked from. It is also the only value that results in absolute paths in source unit names being actually interpreted as absolute paths on disk. If the base path itself is relative, it is interpreted as relative to the current working directory of the compiler."_

Reply:
See also item 5 in my writeup at the start of this PR